installer/ovatools/ovfenv/main.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016-2017 VMware, Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"encoding/xml"
+	"fmt"
+	"os"
+	"strings"
+
+	"gopkg.in/urfave/cli.v1"
+
+	"github.com/vmware/vmw-guestinfo/rpcvmx"
+	"github.com/vmware/vmw-guestinfo/vmcheck"
+
+	"github.com/vmware/vic/pkg/version"
+)
+
+type environment struct {
+	Platform   map[string]string
+	Properties map[string]string
+}
+
+func main() {
+
+	app := cli.NewApp()
+	app.Usage = "Fetch OVF environment information"
+	app.Version = version.GetBuild().ShortVersion()
+	app.Flags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "key, k",
+			Value: "",
+			Usage: "Get single OVF property",
+		},
+	}
+
+	app.Action = func(c *cli.Context) error {
+
+		// Check if we're running inside a VM
+		isVM, err := vmcheck.IsVirtualWorld()
+		if err != nil {
+			fmt.Printf("error: %s\n", err.Error())
+			os.Exit(-1)
+		}
+
+		// No point in running if we're not inside a VM
+		if !isVM {
+			fmt.Println("not living in a virtual world... :(")
+			os.Exit(-1)
```

Comment:
Just old habits, `-1` actually returns `255` as exit code, but serves as a reminder in the code that it's an unexpected error condition.